Source/Samples/Sections/Widgets/EntrySection.cs

```diff
@@ -66,5 +69,103 @@ public EntrySection()
             return ("Invisible text entry:", entry);
         }
 
+        public (string, Widget) CreateSearchEntry()
+        {
+            var entry = new Entry();
+            entry.PlaceholderText = "Search";
+            entry.SetIconFromIconName(EntryIconPosition.Primary, "edit-find");
+            entry.SetIconFromIconName(EntryIconPosition.Secondary, "edit-clear");
+
+            entry.IconRelease += (o, args) =>
+            {
+                switch (args.P0)
+                {
+                    case EntryIconPosition.Primary:
+                        ApplicationOutput.WriteLine(o, "Clicked Search Icon");
+                        break;
+
+                    default:
+                        entry.Text = string.Empty;
+                        ApplicationOutput.WriteLine(o, "Clicked Clear Icon");
+                        break;
+                }
+            };
+
+            return ("Search entry:", entry);
+        }
+
+        public (string, Widget) CreateProgressEntry()
+        {
+            var entry = new Entry();
+            entry.PlaceholderText = "Progress Entry";
+            entry.ProgressFraction = 0f;
+            entry.MaxLength = 20;
+
+            entry.Changed += (sender, e) => entry.ProgressFraction = (float)entry.Text.Length / entry.MaxLength;
+
+            return ("Progress entry:", entry);
+        }
+
+        Entry DemoEntry = new Entry();
+
+        public (string, Widget) CreateCompletionEntry()
+        {
+            // create completion object and assign it to entry
+            var completion = new EntryCompletion();
+            DemoEntry.Completion = completion;
+
+            // create values store
+            var store = new ListStore(typeof(string));
+            store.AppendValues("An example to search for");
+            store.AppendValues("Better example");
+            store.AppendValues("Better and bigger example");
+            store.AppendValues("Some other example");
+
+            // create tree column
+            var column = new TreeViewColumn();
+            column.Reorderable = false;
+            column.SortIndicator = false;
+            column.Clickable = true;
+            column.SortColumnId = 0;
+
+            // create cells
+            var cell = new CellRendererText();
+            column.PackStart(cell, true);
+
+            // set cell functions
+            column.SetCellDataFunc(cell, (TreeViewColumn tree_column, CellRenderer cellRenderer, ITreeModel tree_model, TreeIter iter) =>
+            {
+                var m = tree_model.GetValue(iter, 0);
+                (cellRenderer as CellRendererText).Text = (string)m;
+            });
+                                          
+            // model filter
+            var filter = new TreeModelFilter(store, null);
+            filter.VisibleFunc = new TreeModelFilterVisibleFunc(FilterTree);
```

Comment:
Please use a lambada or an anonymous delegate instead of a separate method for this.